java/src/main/java/com/adhoc/flight/client/AdhocFlightClient.java

```diff
@@ -251,11 +255,70 @@ public void runQuery(String query, HeaderCallOption headerCallOption, RowProcess
         AutoCloseables.close(stream);
     }
 
+	public void runQuery(String query, HeaderCallOption headerCallOption, boolean printResults)
+			throws Exception {
+    	
+    	try(FlightStream flightStream = getStream(
+			getInfo(query, bearerToken, headerCallOption), bearerToken, headerCallOption);
+    			BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE)) {
+    		
+    		Field[] fields = flightStream.getSchema().getFields().stream().toArray(Field[]::new);
+    		int columns = fields.length;
+    		
+    		if (!printResults)
+    			return;
+    		
+    		TinyIntVectorProcessor tinyIntVectorProcessor = new PrintTinyIntVectorProcessor();
+    		
+    		while (flightStream.next()) {
+    			VectorSchemaRoot vectorSchemaRoot = flightStream.getRoot();
+    			int rows = vectorSchemaRoot.getRowCount();
+    			
+    			for (int row = 0; row < rows; row++) {
+    				TinyIntVector[] tinyIntVectors = new TinyIntVector[columns];
```

Comment:
This seems wrong. Isn't the data already in vectors? Why are we copying?

Reply:
Solved in the new version.